### ISSUE TITLE
disable battery optimization problem

### ISSUE THREAD

**Issue by paulokenobi** [YTVanced/VancedMicroG#84]
I have a problem. There is a message always active in the notification area "Keep microG in background - Tap there to disable battery optimization ..." I have done this several times, entered the settings and disabled the battery optimization for microG and Vanced, but whenever I restart the smartphone the optimization is automatically enabled. I use MIUI 12 on the Redmi Note 8 Pro. Has anyone been there?

**Comment by XaoZlo** [comment]
Reboot helped me

**Comment by ywalia01** [comment]
Just hold it and completely turn off notifications for the app :)

**Comment by ezekiel747** [comment]
Not sure why this was closed.
Turning off notifications for microG is a workaround, but the root cause is not solved. 
In my case i can't even find microG in the apps list in battery optimization section.

**Comment by X1nto** [comment]
> Not sure why this was closed.
> Turning off notifications for microG is a workaround, but the root cause is not solved.
> In my case i can't even find microG in the apps list in battery optimization section.

unhide app from the launcher and disable battery optimisations

**Comment by ezekiel747** [comment]
> unhide app from the launcher and disable battery optimisations

how do i unhide the app from the launcher? Thanks!

**Comment by X1nto** [comment]
> > unhide app from the launcher and disable battery optimisations
> 
> how do i unhide the app from the launcher? Thanks!

It's in microg settings, you can access microg settings using manager

**Comment by ezekiel747** [comment]
Oh, thanks a lot. I wasn't aware of such option.
For anyone having trouble finding the option to un-hide:
Vanced Manager -> expand Vanced microG part -> "open" icon (opens microG settings) -> toggle off "hide icon from launcher".
After this, microG should be found under "Battery optimization" section (make sure the visibility filter is on "all apps")

**Comment by raphaeloreis** [comment]
> Oh, thanks a lot. I wasn't aware of such option. For anyone having trouble finding the option to un-hide: Vanced Manager -> expand Vanced microG part -> "open" icon (opens microG settings) -> toggle off "hide icon from launcher". After this, microG should be found under "Battery optimization" section (make sure the visibility filter is on "all apps")

Fixed my problem

**Comment by valentijnscholten** [comment]
> Oh, thanks a lot. I wasn't aware of such option. For anyone having trouble finding the option to un-hide: Vanced Manager -> expand Vanced microG part -> "open" icon (opens microG settings) -> toggle off "hide icon from launcher". After this, microG should be found under "Battery optimization" section (make sure the visibility filter is on "all apps")

I did this, but after a reboot I'm still getting the notification, even though it is unhidden and battery optimisation is disabled for vanced microg. This is on a OPO8 Pro, Oxygen OS / Android 12.

**Comment by Jarmos-san** [comment]
Stumbled across this thread after perusing through hundreds of threads on Reddit, GitHub and more, so I thought I would share what I did. Be warned it's not a fix and only a temporary "hack" which I'm happy with. 

So I'm pretty sure the battery optimisation is disabled on my device for MicroG. And as suggested I tried rebooting countless times as well only to fail with frustration. So instead I disabled this very specifically mentioned "battery notifications" instead. See screen shot for reference. 

![Screenshot_2023-05-10-18-31-06-687_com android settings](https://github.com/TeamVanced/VancedMicroG/assets/31373860/dc28150e-bd81-4920-aca2-1b847f9ee5f3)

You'll find it under - "Settings > Apps > Vanced microg > Notifications > Battery Notifications". I hope it helps someone else as much as it did for me.
